src/debug/javaDebugProvider.ts

```diff
@@ -0,0 +1,45 @@
+import * as path from "path";
+import * as vscode from 'vscode';
+
+import { IDebugProvider, IDockerParser, IDockerResolver, PortInfo } from './debugInterfaces';
+import * as debugUtils from "./debugUtils";
+import { shell } from '../shell';
+
+// Use the java debugger extension provided by microsoft team for java debugging.
+const defaultJavaDebuggerExtension = "vscjava.vscode-java-debug";
+
+export class JavaDebugProvider implements IDebugProvider {
+    constructor(readonly dockerResolver: IDockerResolver) {
+    }
+
+    public getDebuggerType(): string {
+        return "java";
+    }
+
+    public async isDebuggerInstalled(): Promise<boolean> {
+        if (vscode.extensions.getExtension(defaultJavaDebuggerExtension)) {
+            return true;
+        }
+        const answer = await vscode.window.showInformationMessage(`Please install java debugger extension '${defaultJavaDebuggerExtension}' before debugging.`, "Install Now");

```

Comment:
Possible wording: "Java debugging requires the 'Debugger for Java' extension. Would you like to install it now?"

We need to talk to @squillace about how he feels about baking this into the base extension.  It seems like the sort of thing he'd like to externalise, but I also recognise we need some way to point people at the right debugger for their language (if they don't have one, or one that works remotely, already).

Reply:
Different debugger has different launch configuration definitions. If we allow the user to specify the debugger by themselves, how to get its debug configuration definition in our code?

That's why i don't depend on java debugger in `package.json` directly, but remind the user to install the required extension when he triggered the java debugging feature.